### ISSUE TITLE
Colon ':' char in install path causes load error

### ISSUE THREAD

**Issue by Vlad777** [elastic/logstash#2346]
We are trying to run logstash in Mesos sandbox, which creates paths with colon (:) character, which cause the following error on starting logstash:

Error: Could not find or load main class org.jruby.Main

**Comment by jordansissel** [comment]
Hmm weird. can you provide your logstash command line and config?

**Comment by Vlad777** [comment]
I replicated this scenario with logstgash 1.4.2 with the simplest config (which doesn't matter, since it fails before even starting Jruby). I just renamed by install folder path to include a colon (:) character.

mv logstash-1.4.2 logstash\:1.4.2
cd logstash\:1.4.2
echo "input { stdin {} } output { stdout {} }" > std.conf
bin/logstash -f std.conf
Error: Could not find or load main class org.jruby.Main

**Comment by ph** [comment]
I am not completely familiar with mesos..
I assume this directory is automatically generated by mesos? And could it related to this mesos issue?
https://issues.apache.org/jira/browse/MESOS-1128 last comment talk about path issues and jars.

**Comment by Vlad777** [comment]
That mesos issue covers only the date formatting which was fixed, but there is another part of the path that uses colons as a separator.

It's an issue of the install path for logstash having a (:) character, which can happen without mesos and I've reproduced it easiliy without mesos. In fact, bin/elasticsearch has the same problem if it's installed in a path with a colon (:) character.

If it cannot be fixed, then perhaps this can go into the logstash README, such as "Do not install in a path with colon (:) characters in it."
